proposals/dictionary-expressions.md

```diff
@@ -54,9 +61,44 @@ Choices here would have implications regarding potential syntactic ambiguities,
 
 ## Design Intuition
 
-Intuitively, *dictionary expressions* work similarly to *collection expressions*, except treating a `k:v` element as a shorthand for creating a `System.Collections.Generic.KeyValuePair<TKey, TValue>`.  Many rules for *dictionary expressions* will correspond to existing rules for *collection expressions*, just requiring aspects such as *element* and *iteration types* to be some `KeyValuePair<,>`.
+There are two core aspects to the design of dictionary expressions. 
+
+First is the concept of a `dictionary type`. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`.  The latter requirement ensures that `List<KeyValuePair<int, string>>` is not considered a dictionary type (as its indexer is from `int` to `KeyValuePair<int, string>`), while `Dictionary<TKey, TValue>` would of course be.
```

Comment:
i wanted to up front state what we think a dictionary is, and how we separate them out from existing collection types.  Intuition is both that it is strongly attached to KVP as its element type *AND* it has an indexer.  The latter really is how we say a type is announcing its dictionary-ness.

Reply:
```suggestion
First is the concept of a `dictionary type`. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`. The former requirement ensures that `List<T>` is not considered a dictionary type, with it's `int`-to-`T` indexer. The latter requirement ensures that `List<KeyValuePair<int, string>>` is not considered a dictionary type, with its `int`-to-`KeyValuePair<int, string>` indexer. `Dictionary<TKey, TValue>` passes both requirements.
```